command/issues/list/utils.go

```diff
@@ -0,0 +1,212 @@
+package list
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/deepsourcelabs/cli/deepsource/issues"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+)
+
+type ExportData struct {
+	Occurences []IssueJSON `json:"occurences"`
+	Summary    Summary     `json:"summary"`
+}
+
+type Summary struct {
+	TotalOccurences int `json:"total_occurences"`
+	UniqueIssues    int `json:"unique_issues"`
+}
+
+///////////////////////
+// Filtering utilities
+///////////////////////
+
+// Filters issues based on a path, works for both directories and files
+func filterIssuesByPath(path string, issuesData []issues.Issue) ([]issues.Issue, error) {
+	var filteredIssues []issues.Issue
+	for _, issue := range issuesData {
+		up := ".." + string(os.PathSeparator)
+
+		// get relative path
+		rel, err := filepath.Rel(path, issue.Location.Path)
+		if err != nil {
+			return nil, err
+		}
+
+		// check if the relative path has a parent directory
+		if !strings.HasPrefix(rel, up) && rel != ".." {
+			filteredIssues = append(filteredIssues, issue)
+		}
+	}
+
+	return filteredIssues, nil
+}
+
+// Returns de-duplicated issues based on a unique identifier.
+func getUniqueIssues(identifier string, fetchedIssues []issues.Issue) []issues.Issue {
+	var uniqueIssues []issues.Issue
+	inUnique := make(map[string]bool)
+
+	if identifier == "issue_path" {
+		for _, issue := range fetchedIssues {
+			if _, ok := inUnique[issue.Location.Path]; !ok {
+				inUnique[issue.Location.Path] = true
+				uniqueIssues = append(uniqueIssues, issue)
+			}
+		}
+	}
```

Comment:
Suppose a single file has two different issues reported. Will this keep only one of them?

Reply:
Gotcha. I'll modify the map to use issues as the key, instead of issue path.

Reply:
What if the same issue has been reported in the file for multiple locations?

Reply:
I guess what we can do:
1. Fetch all the issues
2. Check which args are directories and filter issues for them.
3. Check the file args, and see if issues for those files already exist in the issues filtered for directories.
4. If yes, skip adding issues for those files
5. If no, filter issues for those files from the `opts.issuesData` and then append.

What do you think @burntcarrot ?

Reply:
> What if the same issue has been reported in the file for multiple locations?

Since each issue is unique, if the same issue has been reported in the file for multiple locations, then the issues will also have different values for specific fields like line number, etc.

So, `getUniqueIssues` filters out issues which are completely equal (all fields have same values), which means it can handle edge cases like these.